src/kiutils/items/schitems.py

```diff
@@ -213,6 +213,69 @@ def to_sexpr(self, indent=2, newline=True) -> str:
         expression += f'{indents}){endline}'
         return expression
 
+@dataclass
+class BusAlias():
+    """The ``bus_alias`` token defines a bus entry in the schematic
+
+    Documentation:
+        https://gitlab.com/kicad/services/kicad-dev-docs/-/merge_requests/53/diffs
+    """
+
+    name: str = ""
+    """The ``name`` of the bus."""
+
+    members: List[str] = field(default_factory=list)
+    """The list of ``members`` defined in the bus. Note that when you tap out a bus entry
+       from a bus using one these members a label will be created with the selected member name"""
+
+    @classmethod
+    def from_sexpr(cls, exp: list) -> BusAlias:
+        """Convert the given S-Expresstion into a BusAlias object
+
+        Args:
+            - exp (list): Part of parsed S-Expression ``(bus_alias ...)``
+
+        Raises:
+            - Exception: When given parameter's type is not a list
+            - Exception: When the first item of the list is not bus_alias
+
+        Returns:
+            - BusAlias: Object of the class initialized with the given S-Expression
+        """
+        if not isinstance(exp, list):
+            raise Exception("Expression does not have the correct type")
+
+        if exp[0] != 'bus_alias':
+            raise Exception("Expression does not have the correct type")
+
+        if len(exp) != 3:
+            raise Exception("Exactly three items are expected in a bus_alias S-Expression.")
+        
+        if not isinstance(exp[2], list) or exp[2][0] != 'members':
+            raise Exception("bus_alias needs to contain a list of members")
+        
+        object = cls()
+
+        object.name = exp[1]
+        object.members = exp[2][1:]
```

Comment:
This will fail for empty member lists that look like this: `(member )`

We have to check if there are any members first.

Reply:
Python lists allow for index range beyond the end. Here, is a ipython3 shell excerpt for reference.

> Python 3.10.6 (main, Mar 10 2023, 10:55:28) [GCC 11.3.0]
> Type 'copyright', 'credits' or 'license' for more information
> IPython 7.31.1 -- An enhanced Interactive Python. Type '?' for help.
> 
> In [1]: x = ["members"]
> 
> In [2]: x[1:]
> Out[2]: []
> 
> In [3]: x = ["members", "1", "2"]
> 
> In [4]: x[1:]
> Out[4]: ['1', '2']